lib/webhook/issue_wrapper.rb

```diff
@@ -0,0 +1,34 @@
+module RedmineWebhook
+  class IssueWrapper
+    def initialize(issue)
+      @issue = issue
+    end
+
+    def to_hash
+      {
+        :id => @issue.id,
+        :subject => @issue.subject,
+        :description => @issue.description,
+        :created_on => @issue.created_on,
+        :updated_on => @issue.updated_on,
+        :closed_on => @issue.closed_on,
+        :root_id => @issue.root_id,
+        :parent_id => @issue.parent_id,
+        :done_ratio => @issue.done_ratio,
+        :start_date => @issue.start_date,
+        :due_date => @issue.due_date,
+        :estimated_hours => @issue.estimated_hours,
+        :is_private => @issue.is_private,
+        :lock_version => @issue.lock_version,
+        :custom_field_values => @issue.custom_field_values.collect { |value| RedmineWebhook::CustomFieldValueWrapper.new(value).to_hash },
+        :project => RedmineWebhook::ProjectWrapper.new(@issue.project).to_hash,
```

Comment:
Do we need custom wrapper for related class? Why not just `to_json` for related object?

Reply:
IMO this grants more control on data returned and allows to strip unwanted field. Diff:

With custom wrapper

```
      "project": {
        "id": 32,
        "identifier": "front",
        "name": "Frontend team",
        "description": "",
        "created_on": "2012-11-06T14:39:27.104Z",
        "homepage": ""
      },
```

Without wrapper:

```
      "project": {
        "id": 32,
        "name": "Frontend team",
        "description": "",
        "homepage": "",
        "is_public": true,
        "parent_id": 1,
        "created_on": "2012-11-06T14:39:27.104Z",
        "updated_on": "2017-04-03T13:14:05.071Z",
        "identifier": "front",
        "status": 1,
        "lft": 78,
        "rgt": 85,
        "inherit_members": false,
        "landing_page": "https://red.p.ostrovok.ru/projects/front/issues?query_id=321",
        "default_version_id": 644,
        "auto_issue_version": false,
        "lb_assign_role_id": null,
        "associating_cf": null,
        "activity_report_settings": null,
        "default_assigned_to_id": null
      },
```